### ISSUE TITLE
Port e2e tests for the Windows plugin

### ISSUE THREAD

**Issue by venilnoronha** [vmware/docker-volume-vsphere#1658]
The following priorities are identified for porting e2e tests for the Windows plugin.

### P0
* `basic_test.go`
* `volume_access_test.go`
* `volumeproperty_test.go`
* `vsan_test.go`

### P1
* `vmlistener_test.go`
* `refcount_test.go`
* `restart_test.go`

Tests related to `vmgroups` are planned NOT to be ported.

/cc @pshahzeb @shaominchen

**Comment by tusharnt** [comment]
@ashahi1 Can you please take care of the sole P0 test that is remaining?

**Comment by tusharnt** [comment]
This needs to be an epic.

**Comment by shuklanirdesh82** [comment]
Ci integration: this is must need for windows plugin GA.

**Comment by govint** [comment]
@shuklanirdesh82 @ashahi1 do we plan to have a CI with Windows VM,

The script part of the E2E tests need to be able to invoke commands on the ESX host which is what needs to be done (hopefully the e2e tests are uniformly written so changes can be localized). Besides that the rest of the code in Go should just work.

**Comment by shuklanirdesh82** [comment]
@govint we are already having windows VM into our CI. It seems the work was on hold due to #1909 (closed now); this should be unblocked now to finish the required work.

**Comment by shuklanirdesh82** [comment]
@ashahi1 as discussed offline, can you please rerun all required test locally as #1909 closed now? 

Please share your analysis/observation for the next step. If required tests pass then unblock tests against windows vm.

**Comment by tusharnt** [comment]
Plan this week - Manually execute all tests (P0 & P1) and see if that uncovers any issues with Skylight. Once this is done then continue with automation.

**Comment by ashahi1** [comment]
Manually performed all the tests. Hit one issue #2006
